python/MooseDocs/extensions/listing.py

```diff
@@ -166,30 +167,57 @@ class InputListingCommand(FileListingCommand):
     def defaultSettings():
         settings = FileListingCommand.defaultSettings()
         settings['block'] = (None, 'Space separated list of input file block names to include.')
+        settings['remove_block'] = (None, 'Space separated list of input file block names to hide.')
+        settings['remove_param'] = (None, "Space separated list of input parameters to hide. " \
+                                          "Specify the FULL path like: 'Kernels/diffusion/block.'")
         return settings
 
     def extractContent(self, filename):
         """Extract the file contents for display."""
         if self.settings['block']:
-            content = self.extractInputBlocks(filename, self.settings['block'])
+            hit = self.extractInputBlocks(pyhit.load(filename), self.settings['block'])
         else:
-            content = common.read(filename)
-
-        content, _ = common.extractContent(content, self.settings)
+            hit = pyhit.load(filename)
+
+        if self.settings['remove_block']:
+            for block in self.settings['remove_block'].split():
+                node = moosetree.find(hit, lambda n: n.fullpath.endswith(block))
+                if node is None:
+                    msg = "Unable to find block '{}' in {}."
+                    raise exceptions.MooseDocsException(msg, block, filename)
+                node.remove()
+
+        if self.settings['remove_param']:
+            for param in self.settings['remove_param'].split():
+                block = re.sub(r'[^/]+$', '', param)
+                node = moosetree.find(hit, lambda n: n.fullpath + '/' == '/' + block)
+                if node is None:
+                    msg = "Unable to find block '{}' in {}."
+                    raise exceptions.MooseDocsException(msg, block, filename)
+
+                match = node.get(param.replace(block, ''))
+                if match is None:
+                    msg = "Unable to find parameter '{}' in the input file:\n\n{}\n\nCheck for " \
+                          "spelling errors and be sure to specify the FULL path of the " \
+                          "parameter as it appears in the above."
+                    raise exceptions.MooseDocsException(msg, param, str(hit.render()))
+                node.removeParam(param.replace(block, ''))
+
+        content, _ = common.extractContent(str(hit.render()), self.settings)
```

Comment:
@aeslaughter I feel like there's a more clever way to do this, any suggestions?

Reply:
The code used to handle the different settings always searches for a block (in slightly different ways), and issues the same exact exception if not found. Surely, this might be consolidated somehow...

Reply:
I will fix it up a bit, I think it is close.